catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/WhenNfcBrickTest.java

```diff
@@ -0,0 +1,488 @@
+
+		readTagId = dataContainer.addProjectUserVariable(READ_TAG_ID);
+		readTagMessage = dataContainer.addProjectUserVariable(READ_TAG_MESSAGE);
+		numDetectedTags = dataContainer.addProjectUserVariable(NUM_DETECTED_TAGS);
+
+		Sprite sprite = new Sprite("testSprite");
+		Script script = new WhenNfcScript();
+
+		SetVariableBrick setVariableBrickId = new SetVariableBrick(Sensors.NFC_TAG_ID);
+		setVariableBrickId.setUserVariable(readTagId);
+		script.addBrick(setVariableBrickId);
+
+		SetVariableBrick setVariableBrickMessage = new SetVariableBrick(Sensors.NFC_TAG_MESSAGE);
+		setVariableBrickMessage.setUserVariable(readTagMessage);
+		script.addBrick(setVariableBrickMessage);
+
+		ChangeVariableBrick changeVariableBrickNumDetectedTags = new ChangeVariableBrick(new Formula(1), numDetectedTags);
+		script.addBrick(changeVariableBrickNumDetectedTags);
+
+		sprite.addScript(script);
+		project.getDefaultScene().addSprite(sprite);
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+
+		tagDataList = ProjectManager.getInstance().getCurrentSprite().getNfcTagList();
+		NfcTagData tagData = new NfcTagData();
+		tagData.setNfcTagName(getResourcesString(R.string.test_tag_name_1));
+		tagData.setNfcTagUid(NfcHandler.byteArrayToHex(FIRST_TEST_TAG_ID.getBytes()));
+		tagDataList.add(tagData);
+
+		NfcTagData tagData2 = new NfcTagData();
+		tagData2.setNfcTagName(getResourcesString(R.string.test_tag_name_2));
+		tagData2.setNfcTagUid(NfcHandler.byteArrayToHex(SECOND_TEST_TAG_ID.getBytes()));
+		tagDataList.add(tagData2);
+
+		return script;
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		createProjectWithNfcAndSetVariable(PROJECT_NAME);
+		numDetectedTags.setValue(0);
+
+		nfcBrickPosition = 0;
+		setVariableIDPosition = 1;
+		setVariableMessagePosition = 2;
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testBasicLayout() {
+		checkIfBrickAtPositionShowsString(nfcBrickPosition, R.string.brick_when_nfc);
+
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.set_variable_spinner, setVariableIDPosition, READ_TAG_ID);
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.set_variable_spinner, setVariableMessagePosition,
+				READ_TAG_MESSAGE);
+	}
+
+	@Test
+	public void testAddNewTag() {
+		List<String> spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_1),
+				getResourcesString(R.string.test_tag_name_2));
+		List<String> spinnerValuesResourceIdsNotContained = Arrays.asList(THIRD_TEST_TAG_ID, FOURTH_TEST_TAG_ID);
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained, spinnerValuesResourceIdsNotContained);
+
+		UiHardwareTestUtils.gotoNfcFragment(nfcBrickPosition);
+		NfcTagAdapter adapter = UiHardwareTestUtils.getNfcTagAdapter();
```

Comment:
there has to be a nicer way to get to the adapter.... try to get around all the getActivity calls which were never intended for this kind of use in the first place.